src/http/request.cr

```diff
@@ -98,6 +98,19 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    host = @headers["Host"]?
+    return unless host
+    index = host.index(":")
+    index ? host[0...index] : host
```

Comment:
It's not good enough to follow the RFC I'm afraid. Just because the RFC says `Host: 123::0:1` is invalid and should be `Host: [123::0:1]` we still need to handle the former case since it might appear, either due to a bogus or even a malicious client.